sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -438,8 +438,18 @@ export class SessionReceiver {
   async renewSessionLock(): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
     await this._createMessageSessionIfDoesntExist();
+    let receiverName;
+    if (this._messageSession) {
+      receiverName = this._messageSession.receiverName;
+    } else {
+      throw new Error(
+        "AMQP receiver is closed or undefined. Cannot renew lock on non-existing receiver link"
```

Comment:
Wouldn't the `this._throwIfReceiverOrConnectionClosed();` check in line 439 cover this case of receiver being closed/undefined?

Reply:
right, updated